java/src/main/native/CMakeLists.txt

```diff
@@ -15,9 +15,35 @@
 #=============================================================================
 cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
 
+# Use GPU_ARCHS if CMAKE_CUDA_ARCHITECTURES is not defined
+if(NOT DEFINED CMAKE_CUDA_ARCHITECTURES AND DEFINED GPU_ARCHS)
+  if(NOT "${GPU_ARCHS}" STREQUAL "ALL")
+    set(CMAKE_CUDA_ARCHITECTURES "${GPU_ARCHS}")
+  endif()
+endif()
+
+# If `CMAKE_CUDA_ARCHITECTURES` is not defined, build for all supported architectures. If
+# `CMAKE_CUDA_ARCHITECTURES` is set to an empty string (""), build for only the current
+# architecture. If `CMAKE_CUDA_ARCHITECTURES` is specified by the user, use user setting.
+
+# This needs to be run before enabling the CUDA language due to the default initialization behavior
+# of `CMAKE_CUDA_ARCHITECTURES`, https://gitlab.kitware.com/cmake/cmake/-/issues/21302
+if(NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
+  set(CUDF_JNI_BUILD_FOR_ALL_ARCHS TRUE)
+elseif(CMAKE_CUDA_ARCHITECTURES STREQUAL "")
+  unset(CMAKE_CUDA_ARCHITECTURES CACHE)
+  set(CUDF_JNI_BUILD_FOR_DETECTED_ARCHS TRUE)
```

Comment:
I'm unfamiliar with the wiring. I can't seem to find where `CUDF_JNI_BUILD_FOR_DETECTED_ARCHS` gets used.

Reply:
It's used in [the SetGPUArchs module](https://github.com/rapidsai/cudf/blob/branch-0.19/cpp/cmake/Modules/SetGPUArchs.cmake#L38-L43) which is used by the ConfigureCUDA module.